### ISSUE TITLE
Syllable contains no neumes and no <facs>

### ISSUE THREAD

**Issue by JoyfulGen** [DDMAL/Neon#800]
A certain series of actions in Neon created a syllable that contains no neumes and `<facs>` element: 

<img width="398" alt="046r no neume no facs" src="https://user-images.githubusercontent.com/83373378/165772910-b7cef550-b0d9-4338-9a9a-b3748d1618e8.png">

This means that it's completely invisible in Neon and can't be interacted with at all. This empty syllable was definitely created by a series of actions within Neon, because it's not in the straight-outta-omr file. However, the straight-outta-omr file **does** have a syllable with the same ID, but that syllable looks perfectly normal:

<img width="866" alt="046r same syllable before" src="https://user-images.githubusercontent.com/83373378/165773369-cf9702df-3ab0-4b64-913c-ae701e52e070.png">

This problem currently has a relatively simple solution: erase the syllable directly in the MEI file. However, it would be good to prevent it from happening at all, to avoid any confusion.

This is the file before correction:
[046r.mei.zip](https://github.com/DDMAL/Neon/files/8583559/046r.mei.zip)
 
@ailynn-ang feel free to add any other relevant info!

**Comment by yinanazhou** [comment]
Hi @timothydereuse, the output of the MEI encoding job already has empty syllables. Could you please look into this issue while updating python version?

**Comment by timothydereuse** [comment]
yes, I think this would happen when no neumes are recognized above a syllable. will add this as an issue.

**Comment by JoyfulGen** [comment]
I think there are actually two empty syllable situations going on. There are indeed empty syllables in the straight-outta-OMR file, but I don't think this is the same situation. The empty syllables coming out of OMR have a `<facs>` element, so they have a bounding box. Selecting that bounding box and deleting it makes that syllable disappear. In this situation, however, the empty syllable has no `<facs>`, so it has no bounding box and is effectively completely invisible.

Also, this empty syllable is definitely created within Neon, because it has neumes in it in the straight-outta-OMR file (that's the screenshot I added above). I think I've just encountered this situation myself, so I'll try to figure out what series of actions causes this.

**Comment by yinanazhou** [comment]
Hi @JoyfulGen, could you please send me the file with the empty syllable that's created in Neon? Thanks!

**Comment by JoyfulGen** [comment]
[CH-E_611_046r.mei.zip](https://github.com/DDMAL/Neon/files/8652277/CH-E_611_046r.mei.zip)
Strange syllable is "WHERE"

**Comment by yinanazhou** [comment]
Hi @JoyfulGen, I just found a series of actions that creates empty syllables, but with `<facs>`
For the syllable you labelled with 'WHERE', I found it in the straight-outta-OMR file, and:
1. insert neumes
2. group the neumes into a syllable
3. switch to select by NC and group the ncs after the divLine into a neume
4. click UNDO
This series of actions result in this file:
![image](https://user-images.githubusercontent.com/74484576/167533279-732a39c8-9c76-46bb-a480-7ada2d4d1fe7.png)

**Comment by yinanazhou** [comment]
> Hi @JoyfulGen, I just found a series of actions that creates empty syllables, but with `<facs>` For the syllable you labelled with 'WHERE', I found it in the straight-outta-OMR file, and:
> 
> 1. insert puncta
> 2. group the neumes into a syllable
> 3. switch to select by NC and group the ncs after the divLine into a neume
> 4. click UNDO
> 
> This series of actions result in this file: ![image](https://user-images.githubusercontent.com/74484576/167533279-732a39c8-9c76-46bb-a480-7ada2d4d1fe7.png)
> 
> This issue might relate to issue #760.

Hi @JoyfulGen, could you please confirm that this series of actions no longer causes empty syllables anymore? It seems like it is fixed in PR #836.

**Comment by JoyfulGen** [comment]
This issue has been fixed! Also, there's a handy MEI Action that allows the user to delete all empty syllables from the get-go. Closing issue.
